ApplicationLibCode/Commands/ApplicationCommands/RicSumoDataFeature.cpp

```diff
@@ -0,0 +1,295 @@
+void SimpleDialog::onAssetsClicked()
+{
+    if ( !isTokenValid() ) return;
+
+    m_sumoConnector->requestAssets();
+    m_sumoConnector->assets();
+
+    label->setText( "Requesting fields (see log for response" );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void SimpleDialog::onCasesClicked()
+{
+    if ( !isTokenValid() ) return;
+
+    QString fieldName = "Drogon";
+    m_sumoConnector->requestCasesForField( fieldName );
+
+    label->setText( "Requesting cases (see log for response" );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void SimpleDialog::onVectorNamesClicked()
+{
+    if ( !isTokenValid() ) return;
+
+    SumoCaseId caseId( "5b783aab-ce10-4b78-b129-baf8d8ce4baa" );
+    QString    iteration = "iter-0";
+
+    m_sumoConnector->requestVectorNamesForEnsemble( caseId, iteration );
+
+    label->setText( "Requesting vector names (see log for response" );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void SimpleDialog::onFindBlobIdClicked()
+{
+    if ( !isTokenValid() ) return;
+
+    SumoCaseId caseId( "5b783aab-ce10-4b78-b129-baf8d8ce4baa" );
+    QString    iteration  = "iter-0";
+    QString    vectorName = "FOPT";
+
+    m_sumoConnector->requestBlobIdForEnsemble( caseId, iteration, vectorName );
+
+    label->setText( "Requesting blob ID for vector name (see log for response" );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void SimpleDialog::onParquetClicked()
+{
+    if ( !isTokenValid() ) return;
+
+    if ( m_sumoConnector->blobUrls().empty() )
+    {
+        onFindBlobIdClicked();
+    }
+
+    if ( !m_sumoConnector->blobUrls().empty() )
+    {
+        m_sumoConnector->requestBlobDownload( m_sumoConnector->blobUrls().back() );
+
+        label->setText( "Requesting blob ID for vector name (see log for response" );
+    }
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void SimpleDialog::onShowContentParquetClicked()
+{
+    if ( m_sumoConnector->blobContents().empty() ) return;
+
+    auto blob = m_sumoConnector->blobContents().back();
+
+    auto content = blob.contents;
+
+    QFile file( "f:/scratch/parquet.parquet" );
+    if ( file.open( QIODevice::WriteOnly ) )
```

Comment:
This is unlikely to work on most machines..

Reply:
Yes, removed this debug code